packages/core/src/scp-cf/destination/destination-from-service.ts

```diff
@@ -0,0 +1,424 @@
+import { createLogger } from '@sap-cloud-sdk/util';
+import {
+  addProxyConfigurationInternet,
+  DecodedJWT,
+  decodeJwt,
+  isIdenticalTenant,
+  ProxyStrategy,
+  proxyStrategy,
+  verifyJwt
+} from '../../util';
+import { IsolationStrategy } from '../cache';
+import { serviceToken, userApprovedServiceToken } from '../token-accessor';
+import { addProxyConfigurationOnPrem } from '../connectivity-service';
+import {
+  getDestinationServiceCredentialsList,
+  getService
+} from '../environment-accessor';
+import { DestinationOptions } from './destination-accessor';
+import { Destination } from './destination-service-types';
+import {
+  alwaysProvider,
+  alwaysSubscriber,
+  DestinationSelectionStrategies,
+  DestinationSelectionStrategy
+} from './destination-selection-strategies';
+import { DestinationsByType } from './destination-accessor-types';
+import {
+  fetchDestination,
+  fetchInstanceDestinations,
+  fetchSubaccountDestinations
+} from './destination-service';
+import { destinationCache } from './destination-cache';
+
+type DestinationOrigin = 'subscriber' | 'provider';
+
+const logger = createLogger({
+  package: 'core',
+  messageContext: 'destination-accessor-service'
+});
+
+interface DestinationSearchResult {
+  destination: Destination;
+  fromCache: boolean;
+  origin: DestinationOrigin;
+}
+
+const emptyDestinationByType: DestinationsByType = {
+  instance: [],
+  subaccount: []
+};
+
+/**
+ * Retrieves a destination with the given name from the Cloud Foundry destination service.
+ * Returns null if no destination can be found.
+ * Requires the following service bindings: destination, XSUAA
+ * By default, selects subscriber over provider and instance over subaccount destinations.
+ *
+ * If the destinations are read from the environment, the jwt will be ignored.
+ *
+ * @param name - The name of the destination to be retrieved.
+ * @param options - Configuration for how to retrieve destinations from the destination service.
+ * @returns A promise returning the requested destination on success.
+ */
+export async function getDestinationFromDestinationService(
+  name: string,
+  options: DestinationOptions
+): Promise<Destination | null> {
+  return DestinationFromService.getDestinationFromDestinationService(
+    name,
+    options
+  );
+}
+
+class DestinationFromService {
```

Comment:
consider if you can find a better name, enterprisy manager, retriever, accesor etc.